### ISSUE TITLE
Adding same item, but on different page to the same cart is duplicated.

### ISSUE THREAD

**Issue by amirulmenjeni** [wojodesign/simplecart-js#496]
I have a page called store.php where there are lots of items shown. People can either add the items instantly (these items are inside the simpleCart_shelfItem) to their cart or click on the item for more information on that particular item which brings them to the next page, product.php. In product.php there's only one item, with add to cart button to it too. But this item is treated as new item, and therefore it seems like it's being duplicated. 

TL;DR version:
1. Click on add to cart button of item A in page 1, added item A on the cart with id of SCI-1.
2. Click on add to cart button of item A in page 2, added item A on the cart with id of SCI-2.
3. How do I make it so that item A of id SCI-1 increased in quantity rather than adding a "new different" item in the cart?

**Comment by amirulmenjeni** [comment]
After I came back to this problem a few days later since after this is posted, I think I came up with a solution.

``` html
 <script>
    simpleCart.bind('beforeAdd', function(item, isNew){

    //if item is new grab the item
    if(isNew){
        alert("this is a new item!"); //confirm its a new item
    }

       //get your own *custom* item id using the DOM
    var new_item_id= document.getElementById("get_item_id").innerHTML;

        //loop each item in your cart
    simpleCart.each(function(item,x){
                //if the id of that new item is the same as any item id in cart
            if(item.get("item-id") == this_item){

                    //increment the "would be duplicated" item
            item.quantity(item.quantity()+1);
            return;
                }
    });

       //remove the newly added item                
    item.remove();

    //cancel add
    return false;
    });
  </script>
```

This doesn't fix the problem entirely, though, since this only work when all the cart items are already loaded when you load a to a new page. If anyone can improve this, or have another better approach, please do share. :)

**Comment by camarraz** [comment]
It's been 4 years... But the solution is simple actually:

First, try to spit out the name of your product on the console... the value has spaces ! ! ! 

This means that the cart is adding the raw values without any processing, therefore the condition to match the name and price will never be true, hence the duplicates.

Solution 1 (tedious):  Trim spaces of all values prior to adding to the cart.

**Solution 2 (preferred): Don't add any spaces between the values and the html tags !**
